packages/cmr-client/src/CMRSearchConceptQueue.ts

```diff
@@ -84,23 +78,28 @@ export class CMRSearchConceptQueue {
    * This does not remove the object from the queue.  When there are no more
    * items in the queue, returns 'null'.
    *
-   * @returns {Promise<Object>} an item from the CMR search
    */
-  async peek(): Promise<unknown> {
+  async peek(): Promise<T> {
     if (this.items.length === 0) await this.fetchItems();
+    if (this.items[0] === null) {
+      return null as unknown as T;
+    }
     return this.items[0];
   }
 
   /**
    * Remove the next item from the queue
    *
    * When there are no more items in the queue, returns `null`.
-   *
-   * @returns {Promise<Object>} an item from the CMR search
    */
-  async shift(): Promise<unknown> {
+  async shift(): Promise<T | null> {
     if (this.items.length === 0) await this.fetchItems();
-    return this.items.shift();
+    const item = this.items.shift();
+    // eslint-disable-next-line lodash/prefer-is-nil
+    if (item === null || item === undefined) {
```

Comment:
Is there a reason not to use isNil? Just avoiding an import?

Reply:
Because isNil doesn't work properly with ts-check, in that it doesn't act as a type-guard.      I could work around that in type annotations, but that feels even less maintainable.